app/patients/edit/controller.js

```diff
@@ -547,6 +548,14 @@ export default AbstractEditController.extend(BloodTypes, ReturnTo, UserSession,
     });
   },
 
+  beforeUpdate: function() {
```

Comment:
@gnowoel instead of validating through the model, you should just move the uniqueness code here.  Also, it should only run when the model is new (eg `if (this.get('model.isNew')`

The problem with the code the way it is right now if the model is invalid for other reasons (say for example the user didn't enter a first name, it would change the id).